lib/activity.js

```diff
@@ -36,79 +18,18 @@ class ActivityEvent {
     this.message = message
     this.id = id
   }
-
-  /**
-   * @param {String} line
-   * @returns {ActivityEvent}
-   */
-  static fromLogLine (line) {
-    const { timestamp, text } = parseLog(line)
-    const { type, message, source, id } = JSON.parse(text)
-    return new ActivityEvent({
-      timestamp,
-      type,
-      source,
-      message,
-      id
-    })
-  }
-}
-
-class Activity {
-  constructor (activityPath, logs) {
-    this.activityPath = activityPath
-    this.logs = logs
-  }
-
-  /**
-   * @returns {Promise<ActivityEvent[]>}
-   */
-  async get () {
-    return (await fs.readFile(this.activityPath, 'utf-8'))
-      .trim()
-      .split('\n')
-      .filter(Boolean)
-      .map(line => ActivityEvent.fromLogLine(line))
-  }
-
-  /**
-   * @param {Object} [options]
-   * @param {AbortSignal} [options.signal]
-   * @param {number} [options.nLines=10]
-   * @returns {AsyncGenerator<ActivityEvent>}
-   */
-  async * follow ({ signal, nLines = 10 } = {}) {
-    const tail = new Tail(this.activityPath, {
-      nLines,
-      useWatchFile: platform() === 'win32'
-    })
-    for await (const [line] of on(tail, 'line', { signal })) {
-      yield ActivityEvent.fromLogLine(line)
-    }
-  }
-
-  createWriteStream (source) {
-    const addMetaStream = new AddMetaStream({ source })
-    addMetaStream
-      .pipe(new SerializeStream())
-      .pipe(this.logs.createWriteStream(this.activityPath))
-    return addMetaStream
-  }
-
-  async maybeCreateActivityFile () {
-    await maybeCreateFile(this.activityPath)
-  }
 }
 
-const formatActivityObject = ({ type, message, timestamp }) => {
-  return formatLog(`${type.toUpperCase().padEnd(5)} ${message}`, {
-    timestamp,
-    pretty: true
-  })
+const formatActivityObject = ({ type, message }) => {
+  return `${type.toUpperCase().padEnd(5)} ${message}`
+    .trimEnd()
+    .split(/\n/g)
+    .map(line => `[${(new Date()).toLocaleString()}] ${line}`)
+    .join('\n') + '\n'
 }
 
 module.exports = {
+  activities: new EventEmitter(),
```

Comment:
@bajtos do you know what is best practice around typescript and event emitters? It would be cool to type this with `ActivityEvent`

Reply:
In the past, I think I was using the solution described here: https://stackoverflow.com/a/39145058/69868

```ts
declare interface MyClass {
    on(event: 'hello', listener: (name: string) => void): this;
    on(event: string, listener: Function): this;
}

class MyClass extends events.EventEmitter {
    emitHello(name: string): void {
        this.emit('hello', name);
    }
}
```

Now we need to figure out how to do this with JSDoc comments. I'll give this a try and open a PR if I find a solution.

Reply:
See https://github.com/filecoin-station/core/pull/134#discussion_r1203948272 - that may be the easiest solution.

Reply:
> I'll give this a try and open a PR if I find a solution.

I take that back. Let's look into this after we reach the final design and this PR is landed.